spyderlib/widgets/sourcecode/syntaxhighlighters.py

```diff
@@ -414,6 +414,20 @@ def highlightBlock(self, text):
                     
             match = self.PROG.search(text, match.end())
  
+        # Make blank space less apparent by setting the foreground alpha.
+        # This only has an effect when "Show blank space" is turned on.
+        re_blank = re.compile("\s+")
+        match = re_blank.search(text, offset)
+        while match:
+            start, end = match.span()
+            start = max([0, start+offset])
+            end = max([0, end+offset])

```

Comment:
`start+offset` and `end+offset` are negative when a line in a multiline string starts with spaces.

That code is copied from the loop earlier in the function. Interpreting the code (and some experimentation) taught me that `highlightBlock()` is called for each line of text, which might be part of a multiline string. The relevant quotes are temporarily prefixed to the line. The number of added characters is tracked by the negative valued `offset` variable.